### DISCUSSION TITLE
OpenAI extension does not accept any setting for over 200 tokens

### DISCUSSION THREAD

**Discussion by SatoshiReport** [oobabooga/text-generation-webui#3134] [General]
@matatonic

OpenAI extension does not accept any setting for over 200 tokens.  I had to change the default in script.py to get this to work.  max_new_tokens and new_tokens both did not work.

Thanks for all your great work, making this.

**Comment by matatonic** [comment]
Can you update your install and try this again? I think this issue is fixed in the current release.

**Reply by SatoshiReport** [reply]
I updated last night but will try again, thanks

**Reply by SatoshiReport** [reply]
While we are talking, when you load from the command line it says "Loading the extension "openai"..." but it never goes to "loaded" - that threw me for a while, thinking it was not working and then I learned that it doesn't matter.

**Reply by matatonic** [reply]
Good point, I will update the language, thanks.

**Reply by matatonic** [reply]
I think you must have just missed it by minutes, the fix wouldn't be in script.py anymore if you had caught it a little later.
Sorry for any inconvenience this caused, it was fixed a few weeks back but was stuck in a pull request.

**Reply by SatoshiReport** [reply]
Last point, with TheBloke/Wizard-Vicuna-13B-Uncensored-GPTQ with NO "extension -exllama" I get the full text with 2000 tokens set (it is really slow though).  However with 30B and "extension -exllama" set it stops half way through with openai.  webui and websockets however gives the full text with 30B.

**Reply by SatoshiReport** [reply]
Thanks for resolving the token issue.  So once I get the new code will it default to 2000 or should I set it with max_tokens?

**Reply by matatonic** [reply]
max_token is to set how many tokens to generate, not the context size. The context sized can't be changed in the OpenAI API

**Reply by matatonic** [reply]
Can you give me an example of what's not working? Ie. can you share the code? I'm testing now with TheBloke/Wizard-Vicuna-13B-Uncensored-GPTQ .

**Reply by matatonic** [reply]
I was just testing via the streaming chat api and I had no problems getting the context over 2000 tokens.
llm-text-generation-webui-1  | RuntimeError: start (0) + length (2049) exceeds dimension size (2048).
llm-text-generation-webui-1  | Output generated in 4.11 seconds (69.13 tokens/s, 284 tokens, context 1765, seed 1737398157)

**Reply by SatoshiReport** [reply]
Apologies, I edited my comment a few minutes after posting but it was too late.

**Reply by SatoshiReport** [reply]
Going back to the original issue, I downloaded the latest code, and it still stops after 200 tokens.    Here is how I set my parameter list in R.

```
max_tokens <- 2048
model_tokens <- round(max_tokens*0.9 - tik(content), 0) 
parameter_list <- list(messages = messages, stream = stream, max_new_tokens = model_tokens)
```

What should I be doing to get over 200 tokens?  Thanks

**Reply by SatoshiReport** [reply]
got it!  Used max_tokens.  Thanks for all your help
